libvirt/resource_libvirt_domain.go

```diff
@@ -748,3 +759,36 @@ func newDiskForCloudInit(virConn *libvirt.VirConnection, volumeKey string) (defD
 
 	return disk, nil
 }
+
+func getDomainInterfaces(domain *libvirt.VirDomain) ([]libvirt.VirDomainInterface, error) {
+
+	// get all the interfaces using the qemu-agent, this includes also
+	// interfaces that are not attached to networks managed by libvirt
+	// (eg. bridges, macvtap,...)
+	interfaces := getDomainInterfacesViaQemuAgent(domain, true)
+	if len(interfaces) > 0 {
+		// the agent will always return all the interfaces, both the
+		// ones managed by libvirt and the ones attached to bridge interfaces
+		// or macvtap. Hence it has the highest priority
+		return interfaces, nil
+	}
```

Comment:
Yes, `qemu-agent` is authoritative about the IP addresses. Unfortunately I didn't find a way to figure out if  `qemu-agent` is installed into the VM. No error is raised when there's no agent, an empty list of interfaces is returned. That's why the code has still to use the `ListAllInterfaceAddresses` API in certain cases.